### PR TITLE
Core/Text: Add support for BroadcastTextID in creature_text table.

### PR DESCRIPTION
**Pull request by sucofog**
**Changes proposed:**

Core/Text: Add support for BroadcastTextID in creature_text table.

**Target branch(es):** Master branch.

**Issues addressed:** Not adressed, add full support

**Tests performed:** Tested by me and working.

**Additional Info:**

So creature_text is asynchronous with broadcast_text and locales_broadcast_text.
SQL added with BroadcastTextID (more updates to come to BroadcastTextID)

More updates to come to BroadcastTextID, becouse some BroadcastTextID are not updated yet in creature_text with the proper broadcast_text entry's are remain BroadcastTextID = 0.

Thanks to @HeartWell for initial commit support and to https://github.com/pak3935 @pak3935 for pointing out and add full support.

### CODE REVIEW

**Comment by ShinDarth** [comment]
@sucofog why are you dropping and re-creating the entire `creature_text` ?  instead It's better to update i

File: src/game/Texts/CreatureTextMgr.cpp [hunk 1]
```diff
@@ -40,20 +41,21 @@ class CreatureTextBuilder
 class PlayerTextBuilder
 {
     public:
-        PlayerTextBuilder(WorldObject* obj, WorldObject* speaker, ChatMsg msgtype, uint8 textGroup, uint32 id, uint32 language, WorldObject const* target)
-            : _source(obj), _talker(speaker), _msgType(msgtype), _textGroup(textGroup), _textId(id), _language(language), _target(target)
+		PlayerTextBuilder(WorldObject* obj, WorldObject* speaker, uint8 gender, ChatMsg msgtype, uint8 textGroup, uint32 id, uint32 language, WorldObject const* target)
+			: _source(obj), _talker(speaker), _gender(gender), _msgType(msgtype), _textGroup(textGroup), _textId(id), _language(language), _target(target)
```
Comment (ShinDarth) [inline src/game/Texts/CreatureTextMgr.cpp]: indentation

File: src/game/Texts/CreatureTextMgr.cpp [hunk 2]
```diff
@@ -40,20 +41,21 @@ class CreatureTextBuilder
...
         {
-            std::string const& text = sCreatureTextMgr->GetLocalizedChatString(_source->GetEntry(), _textGroup, _textId, locale);
+            std::string const& text = sCreatureTextMgr->GetLocalizedChatString(_source->GetEntry(), _gender, _textGroup, _textId, locale);
 
             return ChatHandler::BuildChatPacket(*data, _msgType, Language(_language), _talker, _target, text, 0, "", locale);
         }
 
         WorldObject* _source;
         WorldObject* _talker;
+		uint8 _gender;
```
Comment (ShinDarth) [inline src/game/Texts/CreatureTextMgr.cpp]: indentation

File: src/game/Texts/CreatureTextMgr.cpp [hunk 3]
```diff
@@ -85,17 +87,17 @@ void CreatureTextMgr::LoadCreatureTexts()
...
+        temp.group           = fields[1].GetUInt8();
+        temp.id              = fields[2].GetUInt8();
+        temp.text            = fields[3].GetString();
+        temp.type            = ChatMsg(fields[4].GetUInt8());
+        temp.lang            = Language(fields[5].GetUInt8());
+        temp.probability     = fields[6].GetFloat();
+        temp.emote           = Emote(fields[7].GetUInt32());
+        temp.duration        = fields[8].GetUInt32();
+        temp.sound           = fields[9].GetUInt32();
+		temp.BroadcastTextId = fields[10].GetUInt32();
```
Comment (ShinDarth) [inline src/game/Texts/CreatureTextMgr.cpp]: most of the lines modified here are just adding an extra space before ``=``, please fix that

Also fix indentation of the last line

**Comment by sucofog** [comment]
Yes, thats the problem, that i don't really now how to extract BroadcastTextID's from the new updated table, contact me by gitter private chat.

File: src/game/Texts/CreatureTextMgr.cpp [hunk 4]
```diff
@@ -123,11 +125,14 @@ void CreatureTextMgr::LoadCreatureTexts()
...
-            temp.TextRange = TEXT_RANGE_NORMAL;
-        }
+		if (temp.BroadcastTextId)
+		{
+			if (!sObjectMgr->GetBroadcastText(temp.BroadcastTextId))
+			{
+				sLog->outErrorDb("CreatureTextMgr: Entry %u, Group %u, Id %u in table `creature_text` has non-existing or incompatible BroadcastTextId %u.", temp.entry, temp.group, temp.id, temp.BroadcastTextId);
+				temp.BroadcastTextId = 0;
+			}
+		}
```
Comment (ShinDarth) [inline src/game/Texts/CreatureTextMgr.cpp]: indentation

File: src/game/Texts/CreatureTextMgr.cpp [hunk 5]
```diff
@@ -452,5 +454,19 @@ std::string CreatureTextMgr::GetLocalizedChatString(uint32 entry, uint8 textGrou
...
+	else
+		 baseText = groupItr->text;
+	if (locale != DEFAULT_LOCALE && !bct)
+		{
+			LocaleCreatureTextMap::const_iterator locItr = mLocaleTextMap.find(CreatureTextId(entry, uint32(textGroup), id));
+			if (locItr != mLocaleTextMap.end())
+			 	ObjectMgr::GetLocaleString(locItr->second.Text, locale, baseText);
+		}
+		return baseText;
+}
```
Comment (ShinDarth) [inline src/game/Texts/CreatureTextMgr.cpp]: no newline at the end of file

File: src/game/Texts/CreatureTextMgr.cpp [hunk 6]
```diff
@@ -452,5 +454,19 @@ std::string CreatureTextMgr::GetLocalizedChatString(uint32 entry, uint8 textGrou
...
+		baseText = bct->GetText(locale, gender);
+	else
+		 baseText = groupItr->text;
+	if (locale != DEFAULT_LOCALE && !bct)
+		{
+			LocaleCreatureTextMap::const_iterator locItr = mLocaleTextMap.find(CreatureTextId(entry, uint32(textGroup), id));
+			if (locItr != mLocaleTextMap.end())
+			 	ObjectMgr::GetLocaleString(locItr->second.Text, locale, baseText);
+		}
+		return baseText;
```
Comment (ShinDarth) [inline src/game/Texts/CreatureTextMgr.cpp]: fix indentation please

**Comment by sucofog** [comment]
Test with latests changes please!

File: src/game/Texts/CreatureTextMgr.cpp [hunk 7]
```diff
@@ -253,6 +263,7 @@ uint32 CreatureTextMgr::SendChat(Creature* source, uint8 textGroup, WorldObject
     if (range == TEXT_RANGE_NORMAL)
         range = iter->TextRange;
 
+
```
Comment (ShinDarth) [inline src/game/Texts/CreatureTextMgr.cpp]: extra line

File: src/game/Texts/CreatureTextMgr.cpp [hunk 8]
```diff
@@ -452,5 +463,19 @@ std::string CreatureTextMgr::GetLocalizedChatString(uint32 entry, uint8 textGrou
...
+        baseText = bct->GetText(locale, gender);
+    else
+        baseText = groupItr->text;
+    if (locale != DEFAULT_LOCALE && !bct)
+        {
+            LocaleCreatureTextMap::const_iterator locItr = mLocaleTextMap.find(CreatureTextId(entry, uint32(textGroup), id));
+            if (locItr != mLocaleTextMap.end())
+                ObjectMgr::GetLocaleString(locItr->second.Text, locale, baseText);
+        }
+        return baseText;
```
Comment (ShinDarth) [inline src/game/Texts/CreatureTextMgr.cpp]: indentation

**Pull request merged**